app/src/main/java/com/graeberj/taskman/auth/domain/repository/AuthRepositoryImpl.kt

```diff
@@ -1,25 +1,53 @@
 package com.graeberj.taskman.auth.domain.repository
 
+import com.graeberj.taskman.auth.data.mapper.toLoggedInUser
 import com.graeberj.taskman.auth.data.remote.api.ApiService
 import com.graeberj.taskman.auth.data.remote.dto.LoginRequestDto
-import com.graeberj.taskman.auth.data.remote.dto.LoginResponseDto
 import com.graeberj.taskman.auth.data.remote.dto.RegistrationRequestDto
-import retrofit2.Call
+import com.graeberj.taskman.auth.domain.models.LoggedInUser
+import com.graeberj.taskman.util.Resource
 
 class AuthRepositoryImpl(private val api: ApiService) : AuthRepository {
-    override suspend fun loginUser(loginRequest: LoginRequestDto): Call<LoginResponseDto> {
-        return api.loginUser(loginRequest)
+    override suspend fun loginUser(email: String, password: String): Resource<LoggedInUser> {
+        val loginRequestDto = LoginRequestDto(email, password)
+        return try {
+            val result = api.loginUser(loginRequestDto)
+            Resource.Success(data = result.toLoggedInUser())
+        } catch (e: Exception) {
+            Resource.Error(message = e.toString())
+        }
     }
 
-    override suspend fun registerUser(registrationRequest: RegistrationRequestDto): Call<Void> {
-        return api.registerUser(registrationRequest)
+    override suspend fun registerUser(
+        email: String,
+        password: String,
+        fullName: String
+    ): Resource<Boolean> {
+        val requestBody =
+            RegistrationRequestDto(email = email, fullName = fullName, password = password)
+        return try {
+            api.registerUser(requestBody)
+            Resource.Success(data = true)
+        } catch (e: Exception) {
+            Resource.Error(message = e.toString())
+        }
     }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video I show the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc